src/util.ts

```diff
@@ -55,7 +55,13 @@ export const handlePolling = (
     } = options;
 
     return xhrRequest(uri)
-        .then((r: any) => r.getData())
+        .then((r: any) => {
+            let res = r.getData();
+            if (res === '') {
```

Comment:
this is fragile solution... :/ As you can see handlePooling is designed only to handle pooling through responseBody data... its not for handling response itself.

please do not modify handlePooling method but rather make a copy and set required handling...

you're trying to bend existing solution too much where `handlePooling` is not designed for generic solution...

or you have to completely rewrite handlePooling to handle different xhrRequests properly